packages/compile-solidity/src/index.ts

```diff
@@ -59,8 +59,14 @@ export const Compile = {
   // NOTE: this function does *not* transform the source path prefix to
   // "project:/" before passing to the compiler!
   async sources({ sources, options }: SourcesArgs) {
+    let originalSettings = options.compilers.solc.settings;
     options = Config.default().merge(options);
     options = normalizeOptions(options);
+    // if disableDefaults is set, we don't want to insert
+    // the default optimizer settings
+    if (options.compilers.solc.disableDefaults) {
+      options.compilers.solc.settings = originalSettings;
+    }
```

Comment:
Do we want to extract these lines into a separate function? Or can we borrow `normalizeOptions` for this and move the defaulting there?

(Curious what everyone thinks of this)

Reply:
@gnidan when we talked yesterday I thought you said you'd rather not add this logic to `normalizeOptions` but perhaps I remembered wrong. I certainly can do that!

Reply:
Ah, I just meant that we should check to make sure that every use of `normalizeOptions` is preceded by that `Config.default()...`) business. If so, then there's no reason to do that first statement separately.